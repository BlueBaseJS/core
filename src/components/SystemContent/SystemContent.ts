import { BlueRain, JsonComponentSchema, withBlueRain } from '../../index';
import React from 'react';
import moize from 'moize';

/**
 * Shows system content
 */
const SystemContent = ({ bluerain: BR }: { bluerain: BlueRain }) => {
	const appRoutes = BR.Apps.getAllRoutes();
	// Default Routes
	let systemRoutes: JsonComponentSchema[] = [
		{
			component: 'Route',
			props: {
				path: '/',
				exact: true,
				component: BR.Components.IndexPage
			}
		},
		...appRoutes
	];

	// Dynamically add new routes
	systemRoutes = BR.Filters.run('bluerain.system.routes', systemRoutes);

	// Add 404 at the end
	systemRoutes.push({
		component: 'Route',
		props: {
			component: BR.Components.NotFoundPage
		}
	});

	const schema = {
		component: 'RouterSwitch',
		props: { style: { flex: 1, flexGrow: 1 } },
		children: systemRoutes
	};

	return BR.API.JsonToReact.parse(schema);
};
const MoisedSystemContent = moize.react(SystemContent);

export default withBlueRain(MoisedSystemContent) as React.ComponentType<any>;
