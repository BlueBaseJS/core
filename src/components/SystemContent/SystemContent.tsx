import { Text, View } from '../../getComponent';
import React from 'react';
import { ViewProperties } from 'react-native';

export interface SystemContentProps extends ViewProperties {}

export class SystemContent extends React.PureComponent<ViewProperties> {

	render() {

		return (
			<View {...this.props}>
				<Text>
					ðŸš€ BlueBase System Content!
				</Text>
				{this.props.children}
			</View>
		);
	}
}