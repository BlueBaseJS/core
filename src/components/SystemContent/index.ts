import SystemContent from './SystemContent';
export default SystemContent;
