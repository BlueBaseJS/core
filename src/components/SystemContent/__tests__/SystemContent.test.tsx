import React from 'react';
import { SystemContent } from '../SystemContent';
import TestRenderer from 'react-test-renderer';
import WithProvider from '../../../testing/helpers/WithProvider';

beforeEach(() => {
	jest.resetModules();
});

describe('SystemContent', () => {
	const SystemContentWithProvider = (props: any) => (
		<WithProvider>
			<SystemContent {...props} />
		</WithProvider>
	);

	test(`Snapshot SystemContent component`, () => {
		const component = TestRenderer.create(
			<SystemContentWithProvider />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	test(`Snapshot SystemContent component after complete render`, (done) => {
		const component = TestRenderer.create(
			<SystemContentWithProvider />
		);
		setTimeout(() => {
			const tree = component.toJSON();
			expect(tree).toMatchSnapshot();
			done();
		});
	});
});
