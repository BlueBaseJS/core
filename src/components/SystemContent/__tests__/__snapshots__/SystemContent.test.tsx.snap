// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemContent Snapshot SystemContent component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading
  </Text>
</View>
`;

exports[`SystemContent Snapshot SystemContent component after complete render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Helvetica",
          "fontSize": 16,
          "fontWeight": "400",
        },
        undefined,
      ]
    }
  >
    ðŸš€ BlueBase System Content!
  </Text>
</View>
`;
